### PR TITLE
Cors config demoserver

### PR DESCRIPTION
**Pull request by will-moore**
See https://trello.com/c/Rnzgw72H/4469-demo-server-cors-config
Adding CORS config to demo server according to https://docs.openmicroscopy.org/omero/5.4.0/sysadmins/unix/install-web.html#setting-up-cors

NB: I see that ```templates/omero-web-config-for-webapps.j2``` is also used by nightshade, so maybe this can't be used for config that we want in demo only?

Should I duplicate this whole block, pointing at a new config template?
```
    - name:
      become: yes
      template:
        src: templates/omero-web-config-for-webapps.j2
        dest: "{{ omero_web_basedir }}/config/omero-web-config-for-webapps.omero"
        owner: "root"
        group: "root"
        mode: "u=rw,go=r"
      notify:
        - restart omero-web
```

### CODE REVIEW

**Comment by joshmoore** [comment]
See how the extra config works for ns-web-pub: https://github.com/openmicroscopy/prod-playbooks/blob/master/templates/omero-web-config-for-webapps_ns-web-pub.j2

**Comment by will-moore** [comment]
Like that?

**Comment by will-moore** [comment]
How is this tested?

**Comment by joshmoore** [comment]
> Like that?

https://github.com/openmicroscopy/prod-playbooks/blob/master/nightshade-webclients.yml#L159


> How is this tested?

Probably easiest with `vagrant`: https://github.com/openmicroscopy/prod-playbooks/blob/master/Vagrantfile unless @kennethgillen has a better suggestion.

**Comment by will-moore** [comment]
Sorry - "like that?" was asking if my commit looked OK? https://github.com/openmicroscopy/prod-playbooks/pull/29/commits/9ed91aea79c31c362d878f818eea4ed2c1419e33

**Comment by joshmoore** [comment]
> Sorry - "like that?" was asking if my commit looked OK?

🤦‍♂️ Doh. Sorry. Yes, something like that.

**Comment by manics** [comment]
Looks like a merge went very very wrong on the Vagrantfile, there's several loops without an `end`.

**Comment by kennethgillen** [comment]
@will-moore  - you can make a fresh Vagrantfile if you prefer to move past this for now.

e.g. take out the multi-server loops etc, to have a minimal file like:

(similar to the one used in the Ansible training)


```
Vagrant.configure("2") do |config|
  config.vm.box = "centos/7"
  config.vm.provider "virtualbox" do |vb|
    vb.customize ["modifyvm", :id, "--memory", "2048"]
    config.vm.network "forwarded_port", guest: 80, host: 8080,
      auto_correct: true
    config.vm.network "forwarded_port", guest: 4064, host: 4064,
      auto_correct: true
    config.vm.network "forwarded_port", guest: 4063, host: 4063,
      auto_correct: true
  end

  config.vm.provision "ansible" do |ansible|
    ansible.skip_tags = "lvm"
    ansible.playbook = "nightshade-webclients.yml"
    ansible.galaxy_role_file = "requirements.yml"
  end
end
```

**Comment by will-moore** [comment]
Thanks @kennethgillen that seemed to work:
```
$ vagrant up
...
- openmicroscopy.selinux-utils was installed successfully
    default: Running ansible-playbook...

PLAY RECAP *********************************************************************
```
I'd like to be able to connect to OMERO.web, but ```localhost:4064``` in my browser says "localhost didn’t send any data."

Also tried to ssh but not sure how to use ```SSH auth method: private key```? 
```
ssh vagrant@127.0.0.1
```

**Comment by manics** [comment]
Untested, but changes look fine.

**Comment by joshmoore** [comment]
Apologies, but my attempts at having automated testing got side-tracked on self-signed HTTPS certs. I do plan to get back to it, but I don't think that need to block this PR.

@kennethgillen: I'd defer to you. My assumption (based on code review) is that this will go through fine when we roll out 5.4.2 to demo.o.org. We _may_ need to open a last minute PR against this repo, though. If you're ok with our handling that later this week, I'd say let's go ahead and get this in.

File: templates/omero-web-config-for-cors.j2 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# {{ ansible_managed }}
+# Add web-extension OMERO.web configuration
+# after web already installed and running via role
+
+# CORS
+config append -- omero.web.apps '"corsheaders"'
+config append omero.web.middleware '{"index": 0.5, "class": "corsheaders.middleware.CorsMiddleware"}'
+config append omero.web.middleware '{"index": 10, "class": "corsheaders.middleware.CorsPostCsrfMiddleware"}'
+config set omero.web.cors_origin_allow_all True
```
Comment (kennethgillen) [inline templates/omero-web-config-for-cors.j2]: Add a newline at EOF?

File: templates/omero-web-config-for-webapps.j2 [hunk 2]
```diff
@@ -21,4 +21,4 @@ config append -- omero.web.ui.center_plugins '["Auto Tag", "omero_webtagging_aut
 
 # tagsearch
 config append -- omero.web.apps '"omero_webtagging_tagsearch"'
-config append -- omero.web.ui.top_links '["Tag Search", "tagsearch"]'
+config append -- omero.web.ui.top_links '["Tag Search", "tagsearch"]'
```
Comment (kennethgillen) [inline templates/omero-web-config-for-webapps.j2]: add a newline at EOF?

**Comment by kennethgillen** [comment]
@joshmoore @will-moore 

Okay - 👍  merge away for 5.4.2 on the basis that I'll open a PR if there are any problems at deployment time.

I'll see if there's time to test in vagrant manually in advance of the deployment.

File: templates/omero-web-config-for-cors.j2 (hunk 1 above)
Comment (will-moore) [reply templates/omero-web-config-for-cors.j2]: Done, thx.

**Comment by kennethgillen** [comment]
Thanks @will-moore - 👍 , good to merge.

**Pull request merged**